### PR TITLE
[8.0] use python 3.11 by default

### PR DESCRIPTION
**Pull request by fstagni**

### CODE REVIEW

**Comment by fstagni** [comment]
@andresailer would you have an idea about the doc fails now?

**Comment by andresailer** [comment]
These warnings probably:
```
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/7136/src/DIRAC/FrameworkSystem/private/standardLogging/LogLevels.py:docstring of DIRAC.FrameworkSystem.private.standardLogging.LogLevels.LogLevel.from_bytes:9: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/docs/checkouts/readthedocs.org/user_builds/dirac/checkouts/7136/src/DIRAC/FrameworkSystem/private/standardLogging/LogLevels.py:docstring of DIRAC.FrameworkSystem.private.standardLogging.LogLevels.LogLevel.to_bytes:8: WARNING: Inline interpreted text or phrase reference start-string without end-string.
```

**Comment by andresailer** [comment]
This is an inherited docstring from a builtin(?, enum). I need to remember how to deal with that...

**Comment by fstagni** [comment]
Comes from https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/FrameworkSystem/private/standardLogging/LogLevels.py#L9-L18 and only appearing now in python 3.11

**Comment by andresailer** [comment]
That doesn't fix it, trying locally now...

File: docs/docs.conf [hunk 1]
```diff
@@ -30,6 +30,7 @@ document_private_members = FCConditionsParser
 no_inherited_members =
   DIRAC.Core.Utilities.Graphs.GraphUtilities,
   DIRAC.DataManagementSystem.private.HttpStorageAccessHandler
```
Comment (andresailer) [inline docs/docs.conf]:
```suggestion
  DIRAC.DataManagementSystem.private.HttpStorageAccessHandler,
```

File: docs/docs.conf [hunk 2]
```diff
@@ -30,6 +30,7 @@ document_private_members = FCConditionsParser
...
+  DIRAC.FrameworkSystem.private.standardLogging.LogLevels
```
Comment (andresailer) [inline docs/docs.conf]:
```suggestion
  DIRAC.FrameworkSystem.private.standardLogging.LogLevels,
```

**Comment by andresailer** [comment]
That should do it...
This is what the page looked like before, so ignoring inherited here should make no difference.

https://dirac.readthedocs.io/en/latest/CodeDocumentation/FrameworkSystem/private/standardLogging/LogLevels.html

**Comment by andresailer** [comment]
This is what it would look like with inherited functions.
image

**Comment by fstagni** [comment]
Fine for me, thanks @andresailer 

For everyone else, a few bullets:
- we moved DIRACOS to python3.11 before doing this PR. I think we should have done the contrary.
- merging this would mean that the mamba env used for local developments would need to be recreated. This is easy but @DIRACGrid/dirac-developer should be aware

File: setup.cfg [hunk 3]
```diff
@@ -16,6 +16,7 @@ classifiers =
     Programming Language :: Python :: 3
     Programming Language :: Python :: 3.9
     Programming Language :: Python :: 3.10
+    Programming Language :: Python :: 3.11
```
Comment (chrisburr) [inline setup.cfg:19]: Do we want to bump `python_requires`? Or let Python 3.9 continue on a best-effort basis?

Comment (fstagni) [reply setup.cfg:19]: I think it's fine to continue 3.9.

Comment (chrisburr) [reply setup.cfg:19]: In that case shall we at least leave a pylint (and possibly a pytest) job lying around for Python 3.9

Comment (fstagni) [reply setup.cfg:19]: > In that case shall we at least leave a pylint (and possibly a pytest) job lying around for Python 3.9

Just added them.

**Pull request merged**